src/EditorFeatures/CSharpTest/Completion/CompletionProviders/Snippets/CSharpAutoPropertyCompletionProviderTests.cs

```diff
@@ -0,0 +1,154 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Microsoft.CodeAnalysis.Testing;
+using Roslyn.Test.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.Completion.CompletionProviders.Snippets
+{
+    public abstract class CSharpAutoPropertyCompletionProviderTests : AbstractCSharpSnippetCompletionProviderTests
+    {
+        [WpfFact, Trait(Traits.Feature, Traits.Features.Completion)]
+        public async Task MissingInNamespace()
+        {
+            await VerifyPropertyAbsenceAsync("""
+                namespace Namespace
+                {
+                    $$
+                }
+                """);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.Completion)]
+        public async Task MissingInFilescopedNamespace()
+        {
+            await VerifyPropertyAbsenceAsync("""
+                namespace Namespace;
+
+                $$
+                """);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.Completion)]
+        public async Task MissingInTopLevelContext()
+        {
+            await VerifyPropertyAbsenceAsync("""
+                System.Console.WriteLine();
+                $$
+                """);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.Completion)]
+        public async Task InsertSnippetInClass()
+        {
+            await VerifyDefaultPropertyAsync("""
+                class MyClass
+                {
+                    $$
+                }
+                """);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.Completion)]
+        public async Task InsertSnippetInRecord()
+        {
+            await VerifyDefaultPropertyAsync("""
+                record MyRecord
+                {
+                    $$
+                }
+                """);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.Completion)]
+        public async Task InsertSnippetInStruct()
+        {
+            await VerifyDefaultPropertyAsync("""
+                struct MyStruct
+                {
+                    $$
+                }
+                """);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.Completion)]
+        public virtual async Task InsertSnippetInInterface()
```

Comment:
I think I would prefer just writing this separately for the different prop snippets